pkg/metamorph/client.go

```diff
@@ -258,28 +259,23 @@ func (m *Metamorph) SubmitTransactions(ctx context.Context, txs sdkTx.Transactio
 	}
 
 	// put all transactions together
-	responses, err := m.client.PutTransactions(ctx, in)
-	if err != nil {
-		m.logger.Warn("Failed to submit transactions", slog.String("key", err.Error()))
-		if m.mqClient != nil {
-			for _, tx := range in.Transactions {
-				err := m.mqClient.PublishMarshal(SubmitTxTopic, tx)
-				if err != nil {
-					return nil, err
-				}
-			}
-
-			// parse response and return to user
-			ret := make([]*TransactionStatus, 0)
-			for _, tx := range txs {
-				ret = append(ret, &TransactionStatus{
-					TxID:      tx.TxID(),
-					Status:    metamorph_api.Status_QUEUED.String(),
-					Timestamp: m.now().Unix(),
-				})
-			}
-
-			return ret, nil
+	start := time.Now()
+	const interval = 300 * time.Millisecond
+	maxTimeout := time.Duration(0)
+	if len(in.Transactions) != 0 {
+		maxTimeout = time.Duration(in.Transactions[0].MaxTimeout) * time.Second
+	}
+	var responses *metamorph_api.TransactionStatuses
+	var err error
+	for {
+		responses, err = m.client.PutTransactions(ctx, in)
+		if err == nil {
+			break
+		}
+		slog.String("err", err.Error())
```

Comment:
That is not going to log anything

This would be needed

```go
m.logger.Error("Failed to put transactions", slog.String("err", err.Error()))
```